controllers/topics/create.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+const {insertTopics} = require('../../utils/post');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling here is well done. It checks if the topic already exists and returns an appropriate response.



<!-- This is an auto-generated comment by OSS CodeRabbit -->